src/hashes/blake2b.c

```diff
@@ -342,7 +342,7 @@ int blake2b_process(hash_state *md, const unsigned char *in, unsigned long inlen
       unsigned long fill = BLAKE2B_BLOCKBYTES - left;
       if (inlen > fill) {
          md->blake2b.curlen = 0;
-         XMEMCPY(md->blake2b.buf + left, in, fill); /* Fill buffer */
+         XMEMCPY(md->blake2b.buf + (left % sizeof(md->blake2b.buf)), in, fill); /* Fill buffer */
```

Comment:
I think that `left == BLAKE2B_BLOCKBYTES` situation can happen when you sent 1.chunk 256B + 2.chunk of any size. But if `left == BLAKE2B_BLOCKBYTES` then `fill == 0` in which case we should perhaps avoid completely calling `XMEMCPY`